C#/Botkit/Microsoft.BotKit.Adapters.Slack/SlackAdapter.cs

```diff
@@ -97,7 +98,7 @@ public SlackAdapter(ISlackAdapterOptions options) : base()
         /// </summary>
         /// <param name="activity"></param>
         /// <returns></returns>
-        public async Task<SlackAPI> GetAPIAsync(Activity activity)
+        public async Task<SlackTaskClient> GetAPIAsync(Activity activity)
```

Comment:
I propose refactoring this method as follows:
```
if (Slack != null)
{
    return null;
}

if ((activity.Conversation as dynamic).team == null)
{
   throw new Exception ("Unable to create API based on activity: ", activity);
}

var token = ......
return string.IsNullOrEmpty(token) ? .....
```

This will improve the readability a lot.

Reply:
Apply a similar approach also to this method:
- GetBotUserByTeamAsync